File: Iniciante/1045 - Tipos de Triângulos.py
"""
Autor: Pierre Vieira
Data da submissão: 25/01/2019 01:50:33
"""
linha = input().split(" ")
x, y, z = linha
a = float(x)
b = float(y)
c = float(z)
cont = 0
aux = 0
while cont == 0:
    cont = 1
    if a < b:
        aux = a
        a = b
        b = aux
        cont = 0
    if a < c:
        aux = a
        a = c
        c = aux
        cont = 0
    if b < c:
        aux = b
        b = c
        c = aux
        cont = 0
if a >= b+c:
    print("NAO FORMA TRIANGULO")
else:
    if a**2 == b**2 + c**2:
        print("TRIANGULO RETANGULO")
    if a**2 > b**2 + c**2:
        print("TRIANGULO OBTUSANGULO")
    if a**2 < b**2 + c**2:
        print("TRIANGULO ACUTANGULO")
    if a == b and b == c:
        print("TRIANGULO EQUILATERO")
    if a==b and b != c or a==c and b != c or b==c and a!=c:
        print("TRIANGULO ISOSCELES")

File: C#/Iniciante/1045 - Tipos de Triângulos.cs
/*
* Autor: Pierre Vieira
* Data de submissão: 14/05/2020 13:11:24
*/
using System;
using System.Globalization;
using Microsoft.VisualBasic;

namespace URI {
    class URI {
        static void Main(string[] args)
        {
            string[] linha = Console.ReadLine().Split(' ');
            double a, b, c;
            a = double.Parse(linha[0]);
            b = double.Parse(linha[1]);
            c = double.Parse(linha[2]);
            double[] vetor = {a, b, c};
            Array.Sort(vetor);
            a = vetor[2];
            b = vetor[1];
            c = vetor[0];
            if (a >= b + c || b >= a + c || c >= a + b)
            {
                Console.WriteLine("NAO FORMA TRIANGULO");
            }
            else
            {
                if (a * a == b * b + c * c) {
                    Console.WriteLine("TRIANGULO RETANGULO");
                }
                else if (a * a > b * b + c * c) {
                    Console.WriteLine("TRIANGULO OBTUSANGULO");
                }
                else if (a * a < b * b + c * c) {
                    Console.WriteLine("TRIANGULO ACUTANGULO");
                }
                if (a == b && b == c)
                {
                    Console.WriteLine("TRIANGULO EQUILATERO");
                }
                else if (a == b || a == c || b == c){
                    Console.WriteLine("TRIANGULO ISOSCELES");
                }
            }
        }
    }
}
